server/coordinator/node_picker.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package coordinator
+
+import (
+	"context"
+	"crypto/rand"
+	"math/big"
+
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/pkg/errors"
+)
+
+type NodePicker interface {
+	PickNode(ctx context.Context, registerNodes []cluster.RegisteredNode) (cluster.RegisteredNode, error)
+}
+
+type RandomNodePicker struct{}
+
+func NewRandomNodePicker() NodePicker {
+	return &RandomNodePicker{}
+}
+
+func (p *RandomNodePicker) PickNode(_ context.Context, registerNodes []cluster.RegisteredNode) (cluster.RegisteredNode, error) {
+	// Filter invalid nodes.
+	numOnlineNodes := int64(0)
+	for idx := range registerNodes {
+		if registerNodes[idx].IsOnline() {
+			numOnlineNodes++
+		}
+	}
+	if numOnlineNodes < 0 {
```

Comment:
It can't be negative, do you mean:
```suggestion
	if numOnlineNodes == 0 {
```